python/ray/train/lightning/lightning_predictor.py

```diff
@@ -0,0 +1,111 @@
+                    x, y = batch
+                    y_hat = self.forward(x)
+                    loss = torch.nn.functional.mse_loss(y_hat, y)
+                    self.log("train_loss", loss)
+                    return loss
+
+                def configure_optimizers(self):
+                    optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
+                    return optimizer
+
+
+            batch_size, input_dim, output_dim = 10, 3, 5
+            model = MyModel(input_dim=input_dim, output_dim=output_dim)
+            predictor = LightningPredictor(model=model, use_gpu=False)
+            batch = np.random.rand(batch_size, input_dim).astype(np.float32)
+
+            # Internally, LightningPredictor.predict()  invokes the forward() method
+            # of the model to generate predictions
+            output = predictor.predict(batch)
+
+            assert output["predictions"].shape == (batch_size, output_dim)
+
+
+        .. testoutput::
+            :hide:
+            :options: +ELLIPSIS
+
+    Args:
+        model: The PyTorch Lightning module to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+        use_gpu: If set, the model will be moved to GPU on instantiation and
+            prediction happens on GPU.
+
+
+    """
+
+    def __init__(
+        self,
+        model: pl.LightningModule,
+        preprocessor: Optional["Preprocessor"] = None,
+        use_gpu: bool = False,
+    ):
+        super(LightningPredictor, self).__init__(
+            model=model, preprocessor=preprocessor, use_gpu=use_gpu
+        )
+
+    @classmethod
+    def from_checkpoint(
+        cls,
+        checkpoint: LightningCheckpoint,
+        model: Type[pl.LightningModule],
+        *,
+        preprocessor: Optional[Preprocessor] = None,
+        use_gpu: bool = False,
+        **load_from_checkpoint_kwargs,
+    ) -> "LightningPredictor":
+        """Instantiate the LightningPredictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``LightningTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and preprocessor from.
+                It is expected to be from the result of a ``LightningTrainer`` run.
+            model: A subclass of ``pytorch_lightning.LightningModule`` that
+                defines your model and training logic. Note that this is a class type
+                instead of a model instance.
+            preprocessor: A preprocessor used to transform data batches prior
+                to prediction.
+            use_gpu: If set, the model will be moved to GPU on instantiation and
+                prediction happens on GPU.
+            **load_from_checkpoint_kwargs: Arguments to pass into
+                ``pl.LightningModule.load_from_checkpoint``
+        """
+        model = checkpoint.get_model(model, **load_from_checkpoint_kwargs)
```

Comment:
maybe you don't need to unpack then pack. just pass it like:
```
model = checkpoint.get_model(
    model, load_from_checkpoint_kwargs=load_from_checkpoint_kwargs
)
```
will do?

Reply:
Good idea. Updated!